src/GafferSceneUI/SceneGadget.cpp

```diff
@@ -656,6 +656,86 @@ size_t SceneGadget::objectsAt(
 	return paths.size();
 }
 
+bool SceneGadget::normalAt( const IECore::LineSegment3f &lineInGadgetSpace, V3f &gadgetNormal ) const
+{
+	V3f centerP;
+	ScenePlug::ScenePath centerPath;
+	if( objectAt( lineInGadgetSpace, centerPath, centerP ) )
+	{
+		// Find the normal of the first triangle formed by `pixelPosition` and
+		// left + top, right + top, left + bottom or right + bottom depth samples
+		// where each pair has the same object ID as the center pixel.
+
+		auto viewportGadget = ancestor<ViewportGadget>();
+		const V2f rasterPosition = viewportGadget->gadgetToRasterSpace( lineInGadgetSpace.p1, this );
```

Comment:
After much poking and prodding of the numbers, that turned out to be the solution. Part of that was using a `box` filter for Arnold's depth buffer so we're getting the average of the pixel depth samples instead of the minimum.

Fixed in https://github.com/GafferHQ/gaffer/pull/5630/commits/0854f737b903ecdeda85d02e3a49aa5ddbb31252